src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -238,18 +239,20 @@ private static void AdjustSpans(SegmentedList<ClassifiedSpan> spans, TextSpan wi
     {
         // Create an interval tree so we can easily determine which semantic parts intersect with the 
         // syntactic parts we're looking at.
-        var semanticPartsTree = new SimpleIntervalTree<TClassifiedSpan, TClassifiedSpanIntervalIntrospector>(default, values: null);
+        using var _1 = SegmentedListPool.GetPooledList<TClassifiedSpan>(out var semanticSpans);
 
         // Add all the non-empty semantic parts to the tree.
         foreach (var part in semanticParts)
         {
             if (!getSpan(part).IsEmpty)
             {
-                semanticPartsTree.AddIntervalInPlace(part);
+                semanticSpans.Add(part);
                 finalParts.Add(part);
             }
         }
 
+        var semanticPartsTree = new SimpleBinaryIntervalTree<TClassifiedSpan, TClassifiedSpanIntervalIntrospector>(default, values: semanticSpans);
```

Comment:
tree construction at end.  this will move to the FlatArrayIntervalTree in next pr.